jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/AttachHandler.java

```diff
@@ -522,6 +613,56 @@ static final class teardownHook extends Thread {
 			super("Attach API teardown"); //$NON-NLS-1$)
 		}
 
+		private static void threadJoinHelper(Thread thread, long shutdownDeadlineNs) {
+			long timeout = 100;
+			int retryNumber = 0;
+			for (;;) {
+				long currentNanoTime = System.nanoTime();
+				if (currentNanoTime >= shutdownDeadlineNs) {
+					// already reached shutdownDeadlineNs
+					IPC.logMessage(thread + " already reached shutdownDeadlineNs : currentNanoTime = " + currentNanoTime ); //$NON-NLS-1$
+					break;
+				}
+				long tempTimeout = TimeUnit.NANOSECONDS.toMillis(shutdownDeadlineNs - currentNanoTime);
+				if (tempTimeout == 0) {
```

Comment:
The first test can be eliminated, by just checking `tempTimeout <= 0`.

Reply:
Removed the first test, and updated the second one.